src/shared.ts

```diff
@@ -645,6 +646,19 @@ export async function BlockBlueVerified(user: BlueBlockerUser, config: Config) {
 			return;
 		}
 
+		// Step 0: Check for disallowed words or emojis in usernames.
+		const disallowedWordsWithoutEmptyStrings = config.disallowedWords.filter(word => word !== '');
+		if (disallowedWordsWithoutEmptyStrings.length > 0){
+			// this makes extra sure that emojis are always detected, regardless if they are attached to a word or another string of emojis.
+			const regex = new RegExp(config.disallowedWords.join('|'));
```

Comment:
I think that this is a bad idea? An extension user might want to have their configuration set so that a string of multiple words is blockable but constituent words are not, and removing all whitespace from the display name would not allow that.

You should also not use `formattedUserName` here, because it's a baked string. Instead check against `user.legacy.name` (display name) and/or `user.legacy.screen_name` (username).

Reply:
I've changed it up a bit to preserve whitespaces and by trying to split the names into words; I'm still not sure if this is the best option myself, as there can be probably workarounds against this, but this keeps a similar behavior to what I originally expected, and I'm open to other ideas for what could work best for everyone.

Reply:
I think I would run the regex on the full username? For example: ```ts
const usernameToTest = (user.legacy.name).replace(/(?<= ) +/, '');
if (disallowedWordsAndEmojis.test(usernameToTest) {
  queueBlockUser(...)
}
```
For me the expected behavior would be that strings like "bad word" would trigger a block on a username that is something like "bad word more words" but not "good word more words".

Reply:
the problem with this behavior that you've shown is that it gives out false positives
![image](https://github.com/kheina-com/Blue-Blocker/assets/34368774/dd0c1859-6439-4677-ab51-7ce3219913ec)

Reply:
Ideally, the behavior should be like the current one where it needs a whole word to match for it to block someone

that way we can avoid false positives like this:
![image](https://github.com/kheina-com/Blue-Blocker/assets/34368774/31162645-032a-43fa-9e8f-746a263457a7)

Reply:
Hmmm. Don't these false positives also show up if you split the username into words?

An idea for how to make it so the full word has to match might be to change line 3 in your examples to:
```ts
const disallowedWordsAndEmojis = new RegExp(words.join('(?= |$)|'), 'i');
```
Here's what I get with that lookahead added
![image](https://github.com/kheina-com/Blue-Blocker/assets/14047458/dfc12849-47bb-4e26-9803-a6470bb79651)

Reply:
This might be taking it a bit too far but we could also add logic to account for people putting spaces between letters:
```ts
const disallowedWordsAndEmojis = new RegExp(words.map(word=>word.split('').join(' *')).join('(?= |$)|'), 'i');
```

Reply:
I've merged the latest changes, and after testing it for half an hour; I'm happy with how things are behaving, and I haven't seen any annoying false positives.

Reply:
I added a commit that should also add the lookahead to the last item in the disallowed words list, but this is resolved